src/app/chart/donut/donut.component.ts

```diff
@@ -0,0 +1,144 @@
+import { Component, DoCheck, Input, OnInit } from '@angular/core';
+
+import { cloneDeep, defaults, isEqual, merge } from 'lodash';
+import { ChartDefaults } from '../chart.defaults';
+import { ChartBase } from '../chart.base';
+import { DonutConfig } from './donut-config';
+
+import * as d3 from 'd3';
+
+@Component({
+  selector: 'pfng-chart-donut',
+  templateUrl: './donut.component.html'
+})
+export class DonutComponent extends ChartBase implements DoCheck, OnInit {
+
+  @Input() chartData: any;
+  @Input() config: DonutConfig;
+
+  public donutChartId: any;
```

Comment:
I'm thinking this should be renamed as "id". However, what is the difference between donutChartId and config.chartId? 

We don't have an id attribute for any other patternfly-ng component, so can this just be removed?

Reply:
`config.chartId` is user specified. `donutChartId` is generated from this. This section is actually an interpretation of the patternfly-ng code in `sparkline.component.ts` rather than from angular-patternfly, pasted below for reference:

```
    // Create an ID for the chart based on the chartId in the config if given
    if (this.sparklineChartId === undefined) {
      this.sparklineChartId = 'sparklineChart';
      if (this.config.chartId) {
        this.sparklineChartId = this.config.chartId + this.sparklineChartId;
      }
    }
```

The undefined check suggests users can input sparklineChartId themselves but the API and all the examples in angular-patternfly and patternfly-ng direct users to modify config.chartId. My interpretation drops this check, possibly incorrectly. Maybe there was an intention for users to set these 'new' chart ids in patternfly-ng? I don't know the history here, sorry.

Reply:
Having two IDs is confusing and don't see the need for it. I'd like to clean this up for both the sparkline and donut charts and just use config.chartId.